backend/typescript/services/implementations/camperService.ts

```diff
@@ -18,79 +18,68 @@ const Logger = logger(__filename);
-          camper.campSession,
-          {
-            $push: { campers: newCamper.id },
-          },
-          { runValidators: true },
-        );
-
-        if (!existingCamp) {
-          throw new Error(`CampSession ${camper.campSession} not found.`);
-        }
-      } catch (mongoDbError: unknown) {
-        // rollback user creation
+      newCampers = await MgCamper.insertMany(campers);
+      if (campers.length > 0) {
         try {
-          const deletedCamper: Camper | null = await MgCamper.findByIdAndDelete(
-            newCamper.id,
+          existingCamp = await MgCampSession.findByIdAndUpdate(
+            campers[0].campSession,
+            {
+              runValidators: true,
+            },
           );
-
-          if (!deletedCamper) {
-            throw new Error(`Camper ${newCamper.id} not found.`);
+          if (!existingCamp) {
+            throw new Error(
+              `Camp session ${campers[0].campSession} not found.`,
+            );
           }
-        } catch (rollbackDbError) {
-          const errorMessage = [
-            "Failed to rollback MongoDB camper creation after update camp failure. Reason =",
-            getErrorMessage(rollbackDbError),
-            "MongoDB camper id that could not be deleted =",
-            newCamper.id,
-          ];
-          Logger.error(errorMessage.join(" "));
+        } catch (mongoDbError: unknown) {
+          // rollback camper creation
+          /* eslint-disable no-await-in-loop */
+          for (let i = 0; i < newCampers.length; i += 1) {
+            try {
+              const deletedCamper: Camper | null = await MgCamper.findByIdAndDelete(
+                newCampers[i].id,
+              );
+              if (!deletedCamper) {
+                throw new Error(`Camper ${newCampers[i].id} not found.`);
+              }
+            } catch (rollbackDbError) {
+              const errorMessage = [
+                "Failed to rollback MongoDB camper creation after update camp failure. Reason =",
+                getErrorMessage(rollbackDbError),
+                "MongoDB camper id that could not be deleted =",
+                newCampers[i].id,
+              ];
+              Logger.error(errorMessage.join(" "));
+            }
+          }
+          throw mongoDbError;
         }
-
-        throw mongoDbError;
       }
     } catch (error: unknown) {
       Logger.error(
         `Failed to create camper. Reason: ${getErrorMessage(error)}`,
       );
       throw error;
     }
-
-    return {
-      id: newCamper.id,
-      campSession: camper.campSession,
-      registrationDate: newCamper.registrationDate,
-      hasPaid: newCamper.hasPaid,
-      chargeId: newCamper.chargeId,
-      formResponses: camper.formResponses,
-    };
+    newCamperDTOs = newCampers.map((newCamper) => {
+      return {
+        id: newCamper.id,
+        campSession: newCamper.campSession
+          ? newCamper.campSession.toString()
+          : "",
```

Comment:
Could this be simplified to just `campSession: newCamper.campSession.toString(),` ? Each camper must have an associated campSession to pass all the tests so an empty camp case shouldn't be possible. If this was due to a compile error I don't see one on my end

Reply:
good point i'll update that!